internal/authz/authz_test.go

```diff
@@ -58,3 +73,171 @@ func TestAllRolesExistInFGAModel(t *testing.T) {
+
+	userctx := auth.WithUserSubjectContext(ctx, "user-1")
+
+	// verify the project
+	assert.NoError(t, c.Check(userctx, authz.AuthzRoleAdmin.String(), prj1), "failed to check project")
+
+	// create another project
+	prj2 := uuid.New()
+	assert.NoError(t, c.Write(ctx, "user-1", authz.AuthzRoleAdmin, prj2), "failed to write project")
+
+	// verify the project
+	assert.NoError(t, c.Check(userctx, authz.AuthzRoleAdmin.String(), prj2), "failed to check project")
+
+	// ensure projects for user returns the projects
+	projects, err := c.ProjectsForUser(userctx, "user-1")
+	assert.NoError(t, err, "failed to get projects for user")
+	assert.Len(t, projects, 2, "expected 2 projects for user")
+	assert.Contains(t, projects, prj1, "expected project to be returned")
+	assert.Contains(t, projects, prj2, "expected project to be returned")
+
+	// ensure assignments to project returns the user
+	assignments, err := c.AssignmentsToProject(userctx, prj1)
+	assert.NoError(t, err, "failed to get assignments to project")
+	assert.Len(t, assignments, 1, "expected 1 assignment to project")
+	assert.Equal(t, "user-1", assignments[0].Subject, "expected user to be assigned to project")
+
+	// ensure assignments to project returns the user
+	assignments, err = c.AssignmentsToProject(userctx, prj2)
+	assert.NoError(t, err, "failed to get assignments to project")
+	assert.Len(t, assignments, 1, "expected 1 assignment to project")
+	assert.Equal(t, "user-1", assignments[0].Subject, "expected user to be assigned to project")
+
+	// Delete the user (which also deletes the projects)
+	assert.NoError(t, c.DeleteUser(ctx, "user-1"), "failed to delete user")
+
+	// verify the projects are gone
+	assert.Error(t, c.Check(userctx, authz.AuthzRoleAdmin.String(), prj1), "expected project to be gone")
+	assert.Error(t, c.Check(userctx, authz.AuthzRoleAdmin.String(), prj2), "expected project to be gone")
+
+	// ensure projects for user returns no projects
+	projects, err = c.ProjectsForUser(userctx, "user-1")
+	assert.NoError(t, err, "failed to get projects for user")
+	assert.Len(t, projects, 0, "expected 0 projects for user")
+
+	// ensure assignments to project returns no assignments
+	assignments, err = c.AssignmentsToProject(userctx, prj1)
+	assert.NoError(t, err, "failed to get assignments to project")
+	assert.Len(t, assignments, 0, "expected 0 assignments to project")
+}
+
+func newOpenFGAServerAndClient(t *testing.T) (authz.Client, func()) {
+	t.Helper()
+
+	fgaServerRunMux.Lock()
+
+	cfg := run.MustDefaultConfigWithRandomPorts()
+	cfg.Log.Level = "error"
+	cfg.Datastore.Engine = "memory"
+
+	loggr := logger.MustNewLogger("text", "error")
+	serverCtx := &run.ServerContext{Logger: loggr}
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	go func() {
+		err := serverCtx.Run(ctx, cfg)
+		require.NoError(t, err)
+	}()
+
+	testutils.EnsureServiceHealthy(t, cfg.GRPC.Addr, cfg.HTTP.Addr, nil, false)
```

Comment:
Does an instance of the OpenFGA server get started during these tests?

Reply:
it does!